File: Models/Entities/DailyLog.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Models.Entities
{
    public class DailyLog
    {
        public int Id { get; set; }
        public DateTime Date { get; set; } = DateTime.UtcNow.Date;

        public double TotalKcal { get; set; }
        public double TotalProtein { get; set; }
        public double TotalCarbs { get; set; }
        public double TotalFat { get; set; }

        public ICollection<DailyLogMeal> Meals { get; set; } = new List<DailyLogMeal>();

        public string? ApplicationUserId { get; set; }
        [ForeignKey("ApplicationUserId")]
        public ApplicationUser ApplicationUser { get; set; }
    }

}

File: Client/src/app/shared/models/DailyLog.ts
export interface DailyLog {
    id: number
    date: string;
    meals: DailyLogMeal[];
    totalKcal: number;
    totalProtein: number;
    totalCarbs: number;
    totalFat: number;
}

export interface DailyLogDto {
    date: string;
    meals: DailyLogMeal[];
}

export interface DailyLogMeal {
    mealName: string | null;
    recipes: DailyLogRecipe[];
    totalCalories: number | null;
}

export interface DailyLogRecipe {
    foodId: number;
    name: string | null;
    quantityConsumedInGrams: number;
    imageUrl: string | null;
}